### ISSUE TITLE
A large number of pointermove event listeners cause large canvas to be slow

### ISSUE THREAD

**Issue by matas-bitbybit-dev** [retejs/rete#611]
Hey, first of all, thanks for this amazing library. I noticed that with large scripts on the iPad performance of rete canvas is super fast, and started to investigate what could cause the desktop versions to perform way slower. All of this led me to discover that rete v1.5.2 creates a lot of 'pointermove' listeners by instantiating new Drag class on each node. Correct me if I'm wrong, but one pointermove listener should be enough with some global handler that runs all the functions which need to run.

This is my chrome console screenshot of rete canvas eement event listeners:
<img src="https://user-images.githubusercontent.com/128693211/231963752-d9bdea4a-3aa2-4263-9745-a9c845fcb233.jpg" width="300">

...

<img width="328" alt="Screenshot 2023-04-14 at 09 48 49" src="https://user-images.githubusercontent.com/128693211/231965691-e353a4b6-f6cf-42bc-990e-f18ed54f6581.png">

I suppose this is the problem in node.ts file:

<img src="https://user-images.githubusercontent.com/128693211/231963758-24980586-01ef-4705-b5bc-ea03525aa5ef.jpg" width="300">

If I remember correctly on the iPad pointer moves are not triggered as those touch events are handled differently, but rete works fine on the ipad, so I would suspect reducing the number of those listeners on the desktop could make the canvas run way faster. Let me know if I misunderstood the problem. I guess v2 may have solved it, but I can't migrate yet.

**Comment by Ni55aN** [comment]
> I guess v2 may have solved it

exactly, it was fixed in v2.

Regarding v1, PR is welcomed 🙂
Or I will be able to solve it in about a week.

**Comment by info-bitbybit-dev** [comment]
Hey, I'll look into making a PR on this, but week is definitely ok, there's no rush :) Really not sure if I understand Rete codebase enough to make a good contribution, but I will try :)

**Comment by Ni55aN** [comment]
ok, there is an example how it works in v2
https://github.com/retejs/area-plugin/blob/next/src/drag.ts#L39
https://github.com/retejs/area-plugin/blob/next/src/utils.ts#L16

**Comment by Ni55aN** [comment]
@info-bitbybit-dev please let me know if you haven't started working on it so I can start

**Comment by matas-bitbybit-dev** [comment]
hey @Ni55aN I did not really start, sry, I did hack some code to test my idea, but it's definitely not something that could be used by others. Performance improvement wasn't as 'ground breaking' as I thought, at least with my trial. Also, to my surprise, it seems that on safari canvas just renders faster even on desktop where those event listeners are created :) I think whenever you find time your solution will be better, so feel free to implement it in the way that you find best.
